static/app/components/sidebar/sidebarItem.tsx

```diff
@@ -105,10 +105,13 @@ const SidebarItem = ({
   ...props
 }: Props) => {
   // label might be wrapped in a guideAnchor
-  let labelString = label;
-  if (isValidElement(label)) {
-    labelString = label?.props?.children ?? label;
-  }
+  const labelString =
+    typeof label === 'string'
+      ? label
+      : isValidElement(label)
+      ? (label?.props?.children as string)
```

Comment:
Wouldn't it be better to do a runtime typecheck of `label.props.children`?

Reply:
yeah it would be.

Ideally we just change the label to always be a string. Let me see if i can't fix these guide things

Reply:
agree with Billy